src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,227 @@
+== SQL
+
+SQL allows you to query data stored in `IMap` declaratively.
+
+WARNING: The SQL feature is currently in beta. The compatibility between versions
+is not guaranteed. API might change between versions without notice.
+
+=== Example
+
+Consider that we have an IMap `emp` that contains values of type `Employee`:
+
+[source,java]
+----
+public class Employee implements Serializable {
+    private String name;
+    private int age;
+
+    public Employee(String name, int age) {
+        this.name = name;
+        this.age = age;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getAge() {
+        return age;
+    }
+}
+----
+
+The following code will print names of employees whose age is less than 30:
+
+[source,java]
+----
+try (SqlResult result = hazelcastInstance.getSql().execute("SELECT name FROM emp WHERE age < ?", 30)) {
+    for (SqlRow row : result) {
+        String name = row.getObject(0);
+
+        System.out.println(name);
+    }
+}
+----
+
+=== Querying IMap
+
+==== Names
+
+The SQL service exposes `IMap` objects as tables in the predefined `partitioned`
+schema using exact names. This schema is in the SQL service search path so that
+`IMap` objects could be accessed with or without the schema name.
+
+Schema and table names are case sensitive. For example, the `IMap` with the name
+`employee` can be accessed as `employee` or `partitioned.employee`, but not as
+`Employee`.
+
+==== Fields
```

Comment:
I still have some thoughts that this section is not "simple" enough for a newcomer. Let me suggest a bit of reordering and then adding some examples. I would re-arrange the section as:

1. Start with the sentence "Consider the `Employee` class..." and the table that you have below. If you could explicitly write the `SELECT __key, this, name, age FROM emp` query there, it would be absolutely clear what we're talking about. 
2. Continue with the field metadata extraction logic
3. Rest of the content.

Reply:
I am not sure this is the correct approach. How fields are exposed depends on how data is stored, so we should demonstrate a concrete example after we explained how fields are resolved.

I rearranged the whole paragraph as follows:
1. First, mention the key and the value fields with an example
1. Then explain how nested fields are resolved
1. Finally, show an example of the SQL query where all available fields are accessed

Reply:
Regarding your last comment about "top-level object" and "hidden" terms (I cannot comment it for some reason):
- I removed the whole term "top-level object"
- Also removed mentioning of the hidden fields altogether. Instead, I added the separate section for "SELECT *" queries